libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -274,15 +274,21 @@ void IRGeneratorForStatements::initializeStateVar(VariableDeclaration const& _va
 
 		_varDecl.value()->accept(*this);
 
-		writeToLValue(
```

Comment:
This can be reverted back to its original state once transient state variable initialization is forbidden in the fronted; can be strengthened with an assert.

Reply:
Not quite unfortunately; this same branch is used for assigning to state variables in inline assembly apparently, so we'll have to rely on the frontend checks being sturdy enough.

Reply:
One of the tests was failing - I'll revert the change locally and double check, but yeah, I had initially reverted this and strengthened it with an assert, which then ended up causing a test failure (or two).

Reply:
The code path here should only ever be executed for state variables with an initializing value, so if we hit it anywhere else, that'd seem a bit fishy to me :-).

Reply:
Adding the following assert here: 

`solAssert(_varDecl.referenceLocation() != VariableDeclaration::Location::Transient, "State variables cannot be initialized in place.");`

throws in:

`semanticTests/inlineAssembly/inline_assembly_transient_storage_access_inside_function.sol`

I'll investigate further the code path taken in this particular instance.

Reply:
But yeah, I found this curious when it occurred, but left it for later as I had other kinks to work out, which is why I did not resolve this comment.

Reply:
I see that the assert is there now and changes to `writeToLValue()` seen to be gone. So this has been resolved?

Reply:
Yes.